packages/lesswrong/lib/collections/posts/custom_fields.ts

```diff
@@ -293,6 +293,24 @@ addFieldsDict(Posts, {
     insertableBy: ['members'],
   },
 
+  fmCrosspost: {
+    type: new SimpleSchema({
+      isCrosspost: Boolean,
+      hostedHere: { type: Boolean, optional: true },
+      foreignPostId: { type: String, optional: true },
```

Comment:
If `hostedHere` and `foreignPostId` are meant to be nullable, then they should also have the `nullable: true` property - that's the one that the code-gen for types pays attention to. (I'm not actually sure `optional` does anything.)

Reply:
Same thing for the entire `fmCrosspost` field itself

Reply:
Ah, interesting. We probably need to do a bit of an audit of the schemas then because almost everything in that file has `optional: true` and no `nullable` field.